frontend/app/src/composables/history/filter-paginate.ts

```diff
@@ -27,24 +27,27 @@ interface FilterSchema<F, M> {
  * Creates a universal pagination and filter structure
  * given the required fields, can manage pagination and filtering and data
  * fetching when params change
- * @template T,U,V,W,X
+ * @template T,U,V,S,W,X
  * @param {MaybeRef<string | null>} locationOverview
  * @param {MaybeRef<boolean>} mainPage
  * @param {() => FilterSchema<W, X>} filterSchema
  * @param {(payload: MaybeRef<U>) => Promise<Collection<V>>} fetchAssetData
+ * @param {() => S} defaultCollection
  * @param {{onUpdateFilters?: (query: LocationQuery) => void, extraParams?: ComputedRef<LocationQuery>, customPageParams?: ComputedRef<Partial<U>>, defaultSortBy?: {pagination?: keyof T, pageParams?: (keyof T)[], pageParamsAsc?: boolean[]}}} options
  */
 export const useHistoryPaginationFilter = <
   T extends Object,
   U,
```

Comment:
So I am checking a few things and I think it might be safe to assume that `U extends PaginationRequestPayload<T>` can you check if this breaks anything or not?

Maybe we can make it default to `PaginationRequestPayload<T>`


https://github.com/rotki/rotki/blob/3d879ca26c97b7ae40364c724031848ee698faa2/frontend/app/src/types/common.ts#L1

Reply:
checking it.

Reply:
yes it does, but everywhere, there's additional props added to U, setting it as default doesn't break anything, it only means I have to find a way to either swap U with V, or make V have a default value (because Required type parameters may not follow optional type parameters)

Reply:
done, since we're also setting `V = T`